### PR TITLE
Add diagnostics sample plugin

### PR DESCRIPTION
**Pull request by mmorhun**

### CODE REVIEW

File: diagnostics-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## diagnostics-plugin
+
```
Comment (ashumilova) [inline diagnostics-plugin/README.md]: Theia

File: diagnostics-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+Simple Theai plugin to demonstrate diagnostics features of languages plugin API.
```
Comment (ashumilova) [inline diagnostics-plugin/README.md]: Theai = Theia

Comment (mmorhun) [reply diagnostics-plugin/README.md]: Fixed

**Comment by benoitf** [comment]
seems I forgot to hit "submit review" yesterday

**Comment by mmorhun** [comment]
@benoitf I'll fix it in next PR when have time.

File: diagnostics-plugin/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "watch": "webpack-cli -w --config webpack.config.js",
+        "compile": "webpack-cli --config webpack.config.js",
+
+        "build": "yarn run format-code && yarn run compile && theia:plugin pack"
+      },
+      "engines": {
+        "theiaPlugin": "next"
+      },
+      "theiaPlugin": {
+        "frontend": "dist/diagnostics-plugin-frontend-plugin.js"
```
[the remark this answers is not in the document]
Comment (mmorhun) [reply diagnostics-plugin/package.json]: Just easier to debug.

File: diagnostics-plugin/src/diagnostics-plugin-frontend-plugin.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+ *   - toggle updating diagnostics on type
+ *
+ * Tracks diagnostics changes.
+ */
+
+import * as theia from '@theia/plugin';
+
+let autoDiagnostics: boolean = false;
+
+const disposables: theia.Disposable[] = [];
```
[the remark this answers is not in the document]
Comment (mmorhun) [reply diagnostics-plugin/src/diagnostics-plugin-frontend-plugin.ts]: ok, will do. Just haven't catch up to those changes.

**Comment by mmorhun** [comment]
Here is the PR: https://github.com/eclipse/che-theia-samples/pull/4

**Pull request merged**